pkg/skaffold/deploy/resource/deployment.go

```diff
@@ -44,11 +45,27 @@ func (d *Deployment) Deadline() time.Duration {
 	return d.deadline
 }
 
+func (d *Deployment) Status() *Status {
```

Comment:
If I look at this PR in isolation - I wonder, if we have `Status` returning the pointer to `resource.status` - then we don't really need it. Also, UpdateStatus is superflous, we can just Status public? 

I guess you will introduce an interface? Maybe we should add that now?

Reply:
The reason why `status` is private is because `UpdateStatus` should the only way to update a new status for a resource.  I think it makes more sense to make it `Status` not stored as a pointer.